ldnmessages.md

```diff
@@ -0,0 +1,103 @@
+# LDN Messages Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Main Endpoint
+**/api/ldn/ldnmessages**
+
+Provide access to the ldn messages. List all the LDN messages
+
+A sample can be found at https://api7.dspace.org/server/#/server/api/ldn/ldnmessages
+
+Return codes:
+* 200 OK - if the operation succeed
+* 401 Unauthorized - if you are not authenticated
+* 403 Forbidden - if you are not logged in with sufficient permissions. Only system administrators can access
+
+## Single LDN Message
+**/api/ldn/ldnmessages/<:id>**
+
+```json
+{
+  "id" : "urn:uuid:94ecae35-dcfd-4182-8550-22c7164fe23f",
+  "notificationId" : "urn:uuid:94ecae35-dcfd-4182-8550-22c7164fe23f",
+  "queueStatus" : 1,
+  "queueStatusLabel" : "QUEUE_STATUS_QUEUED",
+  "context" : "7d95587b-8e26-4302-9911-3e0f40fc6e4b",
+  "object" : "7d95587b-8e26-4302-9911-3e0f40fc6e4b",
+  "target" : null,
+  "origin" : 1,
+  "inReplyTo" : null,
+  "activityStreamType" : "Announce",
+  "coarNotifyType" : "coar-notify:EndorsementAction",
+  "queueAttempts" : 0,
+  "queueLastStartTime" : null,
+  "queueTimeout" : "2024-01-03T10:56:41.737+00:00",
+  "notificationType" : "Incoming",
+  "type" : "message",
+  "_links" : {
+    "self" : {
+      "href" : "http://localhost/api/ldn/messages/urn:uuid:94ecae35-dcfd-4182-8550-22c7164fe23f"
+    }
+  }
+}
+```
+
+Status codes:
+* 200 OK - if the operation succeed
+* 401 Unauthorized - if you are not authenticated
+* 403 Forbidden - if you are not logged in with sufficient permissions. Only system administrators can access
+* 404 Not found - if no LDN Message exists with such id
+
+## Creating a new LDN Message
+**POST /api/ldn/ldnmessages**
+
+Creation of LDN Message is not supported via Endpoint
+
+Status codes:
+* 405 Method Not Allowed
+
+## PATCH
+
+Patch of LDN Message is not supported via Endpoint
+
+Status codes:
+* 405 Method Not Allowed
+
+## DELETE
+
+Deletion of LDN Message is not supported via Endpoint
+
+Status codes:
+* 405 Method Not Allowed
+
+## enqueueRetry
+**/api/ldn/ldnmessages/enqueueretry/<:id>**
```

Comment:
@abollin & @steph-ieffam i : Please correct me if I'm wrong, but this doesn't seem to align with our REST Contract? https://github.com/DSpace/RestContract/blob/main/README.md#on-sub-path-of-a-single-resource-endpoint-associations

This looks like it should be an "association" style endpoint, but it's not structured as such.  I don't recall any other endpoints that have a structure of `/api/[category]/[endpoint-name]/[subpath]/[uuid]`. Am I misunderstanding this endpoint?

Reply:
should the path be `**/api/ldn/ldnmessages/<:id>/enqueueretry** ` instead of `**/api/ldn/ldnmessages/enqueueretry/<:id>**` as it currently is?

Reply:
@frabacche : I'd recommending asking [that question](https://github.com/DSpace/RestContract/pull/251#discussion_r1509253793) to @abollini .  I'm not sure I understand this endpoint completely yet.  But, it seems *incorrectly designed* to me.  I think making it `/api/ldn/ldnmessages/<:id>/enqueueretry` seems *more correct*... but I still don't know that it should be a `GET` request either.  It's very odd looking to me and I'm not sure I have a recommendation yet.

Reply:
@tdonohue the GET is an awful typo, sorry for that it should be POST actually and also the URL should be changed as noted.

Reply:
@steph-ieffam : The current implementation in `LDNMessageRestController` is using `GET`.  You can see that in the LDNMessageRestControllerIT in https://github.com/DSpace/DSpace/pull/9268    So, the implementation would also require an update as it appears to be using GET as far as I can tell.